### PR TITLE
[WFCORE-3498][WFCORE-3499] Deprecated and unused API cleanup

### PR DESCRIPTION
**Pull request by ropalka**

### CODE REVIEW

File: controller/src/main/java/org/jboss/as/controller/AbstractAddStepHandler.java [hunk 1]
```diff
@@ -345,10 +333,7 @@ protected void performRuntime(final OperationContext context, final ModelNode op
      * changes to runtime services should override this method or the
      * {@link #performRuntime(OperationContext, org.jboss.dmr.ModelNode, org.jboss.as.controller.registry.Resource)} variant.
      * <p>
-     * To provide compatible behavior with previous releases, this default implementation calls the deprecated
-     * {@link #performRuntime(OperationContext, org.jboss.dmr.ModelNode, org.jboss.dmr.ModelNode, ServiceVerificationHandler, java.util.List)}
-     * method. It then does nothing with the objects referenced by the {@code verificationHandler} and
-     * {@code controllers} parameters passed to that method. Subclasses that overrode that method are encouraged to
+     * Subclasses that overrode that method are encouraged to
      * instead override this one or the {@link #performRuntime(OperationContext, org.jboss.dmr.ModelNode, org.jboss.as.controller.registry.Resource)}
      * variant. <strong>Subclasses that override this method should not call{@code super.performRuntime(...)}.</strong>
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/AbstractAddStepHandler.java]: L335-338 should be removed. None of that previous paragraph is relevant now.

File: controller/src/main/java/org/jboss/as/controller/AbstractBoottimeAddStepHandler.java [hunk 2]
```diff
@@ -167,42 +150,15 @@ protected void performBoottime(OperationContext context, ModelNode operation, Re
      * {@link #performBoottime(OperationContext, org.jboss.dmr.ModelNode, org.jboss.as.controller.registry.Resource)}
      * variant.
      * <p>
-     * To provide compatible behavior with previous releases, this default implementation calls the deprecated
-     * {@link #performRuntime(OperationContext, org.jboss.dmr.ModelNode, org.jboss.dmr.ModelNode, ServiceVerificationHandler, java.util.List)}
-     * variant. It then does nothing with the objects referenced by the {@code verificationHandler} and
-     * {@code controllers} parameters passed to that method. Subclasses that overrode that method are encouraged to
-     * instead override this one instead.
+     * Subclasses that overrode that method are encouraged to instead override this one instead.
      * <strong>Subclasses that override this method should not call{@code super.performBoottime(...)}.</strong>
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/AbstractBoottimeAddStepHandler.java]: L152-154 should be removed. None of that previous paragraph is relevant now.

File: controller/src/main/java/org/jboss/as/controller/RestartParentResourceHandlerBase.java [hunk 3]
```diff
@@ -147,10 +147,7 @@ protected void removeServices(final OperationContext context, final ServiceName
     /**
      * Recreate the parent service(s) using the given model.
      * <p>
-     * To provide compatible behavior with previous releases, this default implementation calls the deprecated
-     * {@link #recreateParentService(OperationContext, PathAddress, org.jboss.dmr.ModelNode, ServiceVerificationHandler)}
-     * method. It then does nothing with the objects referenced by the {@code verificationHandler} parameter passed to
-     * that method. Subclasses that overrode that method are encouraged to instead override this one.
+     * Subclasses that overrode that method are encouraged to instead override this one.
      * <strong>Subclasses that override this method should not call{@code super.recreateParentService(...)}.</strong>
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/RestartParentResourceHandlerBase.java]: L149-151 should be removed. None of that previous paragraph is relevant now.

File: controller/src/main/java/org/jboss/as/controller/RestartParentResourceHandlerBase.java [hunk 4]
```diff
@@ -204,7 +180,6 @@ private void recoverServices(final OperationContext context, final ModelNode inv
         if (parentModel != null && context.revertResourceRestarted(address, this)) {
             try {
                 removeServices(context, serviceName, invalidatedParentModel);
-                recreateParentService(context, address, parentModel, null);
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/RestartParentResourceHandlerBase.java]: I don't think the entire statement should be removed; just the param.

File: controller/src/main/java/org/jboss/as/controller/RestartParentWriteAttributeHandler.java [hunk 5]
```diff
@@ -109,10 +109,7 @@ protected void removeServices(final OperationContext context, final ServiceName
     /**
      * Recreate the parent service(s) using the given model.
      * <p>
-     * To provide compatible behavior with previous releases, this default implementation calls the deprecated
-     * {@link #recreateParentService(OperationContext, PathAddress, org.jboss.dmr.ModelNode, ServiceVerificationHandler)}
-     * method. It then does nothing with the objects referenced by the {@code verificationHandler} parameter passed to
-     * that method. Subclasses that overrode that method are encouraged to instead override this one.
+     * Subclasses that overrode that method are encouraged to instead override this one.
      * <strong>Subclasses that override this method should not call{@code super.recreateParentService(...)}.</strong>
```
Comment (bstansberry) [inline controller/src/main/java/org/jboss/as/controller/RestartParentWriteAttributeHandler.java]: L111-113 should be removed. None of that previous paragraph is relevant now.

**Comment by ropalka** [comment]
Thanks for feedback @bstansberry I'm amending requested changes to previous commit.

**Comment by bstansberry** [comment]
retest this please

**Comment by ropalka** [comment]
retest this please

**Pull request merged**